packages/fuel-gauge/src/coverage-contract.test.ts

```diff
@@ -506,6 +506,23 @@ describe('Coverage Contract', () => {
     expect(result.status).toEqual('success');
   });
 
+  it('supports result type', async () => {
```

Comment:
I added this test because I didn't find any other test testing the encoding/decoding of `Result<T, E>`.